### PR TITLE
New architecture for Octavius

### LINKED ISSUE #85: refactor: Split client_cp_service.proto into objective based services
Description:

The present `client_cp_service.proto` file does not follow the micro-services architecture and should be split into objective-based services. In reference to the diagram provided all the user-facing services can be split into two, `metadata_service` and `job_service` proto files.

The `metadata_service` can have methods like `Post, GetAll, Get` and can be used like `metadata.Get`, etc.
The job_service can have methods like `Logs, ListAll, Execute, Describe, Status` and can be used like `job.Execute`, etc.

The separate files created should also be registered on the gRPC served in the api.go so that the application is complete.

Present Behaviour:

The client_cp_service.proto file contain all the user facing services

Expected Behaviour:

As described above.

### PR DESCRIPTION
**Pull request by littlestar642**
Description:

This branch contains the new architecture we have planned for octavius. The major highlights of this architecture is the fact that is more microservice oriented and the code is completely tested and structured. We plan to move ahead with development when this merges to master.

fixes #85

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -5,7 +5,8 @@ import (
 	"log"
 	"octavius/internal/cli/command"
 	"octavius/internal/cli/config"
-	"octavius/internal/cli/daemon"
+	job_daemon "octavius/internal/cli/daemon/job"
```
Comment (faruqisan) [inline cmd/cli/main.go:8]: do not use camel_case for package name

>Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: cmd/cli/main.go [hunk 2]
```diff
@@ -5,7 +5,8 @@ import (
...
+	metadata_daemon "octavius/internal/cli/daemon/metadata"
```
Comment (faruqisan) [inline cmd/cli/main.go:9]: do not use camel_case for package name

>Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: cmd/executor/main.go [hunk 3]
```diff
@@ -3,10 +3,14 @@ package main
 import (
 	"fmt"
 	"log"
-	"octavius/internal/executor/client"
+	healthClient "octavius/internal/executor/client/health"
```
Comment (faruqisan) [inline cmd/executor/main.go:6]: >Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: cmd/executor/main.go [hunk 4]
```diff
@@ -3,10 +3,14 @@ package main
...
+	jobClient "octavius/internal/executor/client/job"
```
Comment (faruqisan) [inline cmd/executor/main.go:7]: >Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: cmd/executor/main.go [hunk 5]
```diff
@@ -3,10 +3,14 @@ package main
...
+	registrationClient "octavius/internal/executor/client/registration"
```
Comment (faruqisan) [inline cmd/executor/main.go:8]: >Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: internal/cli/client/client.go [hunk 6]
```diff
@@ -1,64 +1,2 @@
+//Package client implements a gRPC client
```
Comment (faruqisan) [inline internal/cli/client/client.go:1]: you can remove this file as it empty.

File: internal/cli/command/root.go [hunk 7]
```diff
@@ -7,7 +8,8 @@ import (
 	"octavius/internal/cli/command/execution"
 	"octavius/internal/cli/command/getlogs"
 	"octavius/internal/cli/command/list"
-	"octavius/internal/cli/daemon"
+	jobDaemon "octavius/internal/cli/daemon/job"
```
Comment (faruqisan) [inline internal/cli/command/root.go:11]: >Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: internal/cli/command/root.go [hunk 8]
```diff
@@ -7,7 +8,8 @@ import (
...
+	metadataDaemon "octavius/internal/cli/daemon/metadata"
```
Comment (faruqisan) [inline internal/cli/command/root.go:12]: >Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns.

https://blog.golang.org/package-names

File: internal/cli/daemon/daemon.go [hunk 9]
```diff
@@ -0,0 +1,2 @@
+//Package daemon implements client side logic for all the cli commands
```
Comment (faruqisan) [inline internal/cli/daemon/daemon.go:1]: you can remove this empty file

File: internal/controller/server/execution/health/execution.go [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+			}
+			log.Info(fmt.Sprintf("session ID: %v, deadline exceeded for executor with %s id", executor.sessionID, id))
+			removeActiveExecutor(activeExecutorMap, id, executor)
+			executor.timer.Stop()
+			return
+		case <-executor.pingChan:
+		}
+	}
+}
+func (e *healthExecution) UpdatePingStatus(ctx context.Context, request *protofiles.Ping, pingTimeOut time.Duration) (*protofiles.HealthResponse, error) {
```
Comment (faruqisan) [inline internal/controller/server/execution/health/execution.go:97]: add white space between L96 and L97, and comment for exported function.

File: internal/controller/server/job_services.go [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type jobServicesServer struct {
+	procExec job.JobExecution
+	idgen    idgen.RandomIdGenerator
+}
+
+// JobServiceServer used to create a new execution context
```
Comment (faruqisan) [inline internal/controller/server/job_services.go:23]: NewJobServiceServer

File: cmd/executor/main.go (hunk 3 above)
Comment (littlestar642) [reply cmd/executor/main.go:6]: alias is needed here coz the naming conflicts with daemon packages

**Pull request merged**